src/Control/Distributed/Process/Internal/CQueue.hs

```diff
@@ -101,7 +109,7 @@ dequeue :: forall m a.
         -> BlockSpec         -- ^ Blocking behaviour
         -> [MatchOn m a]     -- ^ List of matches
         -> IO (Maybe a)      -- ^ 'Nothing' only on timeout
-dequeue (CQueue arrived incoming) blockSpec matchons = mask_ $
+dequeue (CQueue arrived incoming size) blockSpec matchons = mask_ $ decrementJust $
```

Comment:
Oh gosh yes, sorry I didn't read the code properly to realise that `decrementJust` takes the result of evaluating the code it's given as its input. That's fine then. And I agree (I think) about the use of `mask_` here, Edsko's previous comments about uninterruptible masking notwithstanding...